fairscale/optim/adascale.py

```diff
@@ -334,9 +340,16 @@ def zero_grad(self) -> None:
         return self._optimizer.zero_grad()
 
     def state_dict(self) -> Dict:
-        """Proxy function to optimizer, checkpointing needs this."""
+        """ Proxy function to optimizer, checkpointing needs this.
+
+            Note: Do NOT checkpoint in the middle of gradient accumulation since
```

Comment:
`.. note:` ? (see https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html)

Reply:
Great idea! Will address both places!